lightning/mydump/router.go

```diff
@@ -262,115 +278,48 @@ func (p *regexRouterParser) parseFieldExtractor(
 		return errors.Errorf("field '%s' match pattern can't be empty", field)
 	}
 
-	// template is const string, return directly
-	if strings.IndexByte(fieldPattern, '$') < 0 {
-		// check if the const pattern is valid
-		if err := validateFn(fieldPattern); err != nil {
-			return err
-		}
-		p.rule.extractors = append(p.rule.extractors, &constExpander{
-			value:   fieldPattern,
-			applyFn: applyFn,
-		})
-		return nil
-	}
-
 	// check and parse regexp template
-	if err := p.checkSubPatterns(fieldPattern); err != nil {
+	if err := p.checkSubPatterns(rule.pattern, fieldPattern, validateFn); err != nil {
 		return errors.Trace(err)
 	}
-	p.rule.extractors = append(p.rule.extractors, &patExpander{
+	rule.extractors = append(rule.extractors, patExpander{
 		template: fieldPattern,
 		applyFn:  applyFn,
 	})
 	return nil
 }
 
-func (p *regexRouterParser) checkSubPatterns(t string) error {
-	captures, err := p.extractSubPatterns(t)
-	if err != nil {
-		return errors.Trace(err)
-	}
-
-outer:
-	for _, c := range captures {
-		if isNumber(c) {
-			idx, _ := strconv.Atoi(c)
-			if idx > p.rule.pattern.NumSubexp() {
-				return errors.Errorf("sub pattern capture index '%d' out of range", idx)
-			}
-		} else {
-			for _, n := range p.rule.pattern.SubexpNames() {
-				if c == n {
-					continue outer
-				}
-			}
-			return errors.Errorf("invalid named capture '%s', available names are: %s", c, p.rule.pattern.SubexpNames())
+func (p regexRouterParser) checkSubPatterns(pat *regexp.Regexp, t string, validateFn func(string) error) error {
+	subPats := expandVariablePattern.FindAllString(t, -1)
+	if len(subPats) == 0 {
+		// validate const patterns
+		if err := validateFn(t); err != nil {
```

Comment:
`len(subPats) > 0` does not mean `t` isn't a const pattern (`expandVariablePattern` matches `$$`). 

i suggest removing `validateFn` and let `applyFn` do the validation naturally. after all, routing is the first thing being done so delaying the validation won't cause errors being caught too late.

Reply:
There is a little difference. If the pattern is an invalid constant pattern for `type` I think It's better to report an error to user. But I the runtime, we will treat the pattern valid and just skip the file because maybe the file will match another pattern that return valid result.

Reply:
no encountering an unknown file or compression type should result in an error, not skipping the file. use the `ignore` type to skip the file.